flujo/tracing/manager.py

```
@@ -118,11 +118,36 @@ async def _handle_pre_run(self, payload: PreRunPayload) -> None:
             # Finalize the root span
             self._root_span.end_time = time.monotonic()  # Use monotonic time
-            self._root_span.status = "completed"
+            # Determine final status
+            paused = False
+            try:
+                ctx = payload.context or getattr(
+                    payload.pipeline_result, "final_pipeline_context", None
+                )
+                scratch = getattr(ctx, "scratchpad", None) if ctx is not None else None
+                paused = isinstance(scratch, dict) and scratch.get("status") == "paused"
+            except Exception:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)